ddsrouter_core/src/cpp/writer/implementations/rtps/Writer.cpp

```diff
@@ -102,10 +121,19 @@ Writer::~Writer()
 utils::ReturnCode Writer::write_(
         std::unique_ptr<DataReceived>& data) noexcept
 {
-
     // Take new Change from history
     fastrtps::rtps::CacheChange_t* new_change = rtps_writer_->new_change(eprosima::fastrtps::rtps::ChangeKind_t::ALIVE);
 
+    //// Copy source writer GUID into inline_qos
+    // First reserve memory for guidPrefix and EntityId
+    new_change->inline_qos.reserve(fastrtps::rtps::GuidPrefix_t::size + fastrtps::rtps::EntityId_t::size);
+
+    // Copy GuidPrefix (relevant for comparison) to the first GuidPrefix_t::size bytes of inline_qos
+    std::memcpy(new_change->inline_qos.data, data->source_guid.guidPrefix.value, fastrtps::rtps::GuidPrefix_t::size);
```

Comment:
Be careful modifying a value that will be sent in the RTPS message with non standard formats. We have to really check this does not imply any issue

Reply:
yes, to be resolved, this is the main reason why this PR is in draft